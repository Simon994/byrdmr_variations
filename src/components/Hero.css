/* Hero section text and images */

:root {
  --drone-base-width: 429px;
  --drone-base-height: 354px;
  --hero-img-scaling: 1.27;
  --hero-img-scaling-phone: 0.956;
}

.hero-section {
  background: #f5f1ed;
  height: 100vh;
}

.hero-inner-container {
  padding-top: 100px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: inherit;
}

#drone-img {
  width: var(--drone-base-width);
  height: var(--drone-base-height);
  z-index: 10;
}

#shadow-img {
  width: 330px;
  z-index: 10;
  position: relative;
  top: -15px;
}

.hero-inner-container h1 {
  font-family:
    'Archivo-Bold',
    'Helvetica Neue',
    sans-serif;
  font-size: 235px;
  font-weight: normal;
  letter-spacing: 18.8px;
  line-height: 265px;
  min-height: 265px;
  margin: -90px 0 0 0;
  color: white;
}

.hero-inner-container h3,
.hero-inner-container p {
  color: rgba(49, 33, 27, 0.23);
  margin-top: 0;
  text-align: center;
  font-weight: 500;
}

.hero-inner-container h3 {
  font-size: 18px;
  margin-bottom: 10px;
  letter-spacing: 1.29px;
}

.hero-inner-container p {
  font-size: 14px;
  letter-spacing: 1px;
}

@media screen and (max-width: 768px) {
  #drone-img {
    width: calc(var(--drone-base-width) * var(--hero-img-scaling));
    height: calc(var(--drone-base-height) * var(--hero-img-scaling));
  }

  #shadow-img {
    width: calc(330px * var(--hero-img-scaling));
    top: -15px;
  }

  .hero-inner-container h1 {
    font-size: 175px;
    margin: -60px 0 20px 0;
    white-space: nowrap;
    letter-spacing: 13px;
  }
}

@media screen and (max-width: 375px) {
  .hero-inner-container {
    padding-top: 50px;
    overflow: hidden;
  }

  #drone-img {
    width: calc(var(--drone-base-width) * var(--hero-img-scaling-phone));
    height: calc(var(--drone-base-height) * var(--hero-img-scaling-phone));
  }

  #shadow-img {
    width: calc(330px * var(--hero-img-scaling-phone));
    top: -15px;
  }

  .hero-inner-container h1 {
    font-size: 100px;
    margin: -20px 0 10px 0;
    letter-spacing: 4.72px;
    line-height: 104px;
    min-height: 116px;
    min-width: 298px;
  }

  .hero-inner-container h3 {
    margin-bottom: 5px;
    max-width: 282px;
  }
}
