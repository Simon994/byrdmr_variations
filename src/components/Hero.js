import { useState, useEffect } from 'react'

import './Hero.css'
import HeroDesktopNavbar from './HeroDesktopNavbar'
import DropdownMenu from './common/DropdownMenu'
import Drone from '../styles/assets/Drone.png'
import Shadow from '../styles/assets/Shadow.png'

const Hero = () => {
  const breakpointTablet = 768
  const [isDesktop, setDesktop] = useState(window.innerWidth > breakpointTablet)

  const updateMedia = () => {
    setDesktop(window.innerWidth > breakpointTablet)
  }

  useEffect(() => {
    window.addEventListener('resize', updateMedia)
    return () => window.removeEventListener('resize', updateMedia)
  })

  return (
    <>
      <section className='hero-section'>
        <div>
          {isDesktop ? (
            <div>
              <HeroDesktopNavbar/>
            </div>
          ) : (
            <div>
              <DropdownMenu/>
            </div>
          )}
        </div>
        <div className='hero-inner-container'>
          <img
            src={Drone}
            id='drone-img'
            alt='Drone'
          />
          <img 
            src={Shadow}
            id='shadow-img'
            alt='Shadow'
          />
          <h1>F-210</h1>
          <h3>&quot;THE FASTEST DRONE ON THE PLANET&quot;</h3>
          <p>&quot;Theâ€” The Wirecutter</p>
        </div>
      </section>
    </>
  )
}

export default Hero
