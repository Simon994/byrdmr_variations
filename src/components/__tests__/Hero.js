import renderer from 'react-test-renderer'
import Hero from '../Hero'
import { BrowserRouter } from 'react-router-dom'

describe('The Hero component', () => {
  it('renders as expected', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Hero/>
      </BrowserRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
