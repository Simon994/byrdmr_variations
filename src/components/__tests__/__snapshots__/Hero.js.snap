// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Hero component renders as expected 1`] = `
<section
  className="hero-section"
>
  <div>
    <div>
      <nav
        className="desktop-nav"
      >
        <div
          className="left-container"
        >
          <ul>
            <li>
              <a
                aria-current={null}
                className="nav-link"
                href="/store"
                onClick={[Function]}
              >
                STORE
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="nav-link"
                href="/tutorials"
                onClick={[Function]}
              >
                TUTORIALS
              </a>
            </li>
          </ul>
        </div>
        <img
          alt="ByrdMR logo"
          className="byrd-logo"
          src="Logo.png"
        />
        <div
          className="right-container"
        >
          <ul>
            <li>
              <a
                aria-current={null}
                className="nav-link"
                href="/blog"
                onClick={[Function]}
              >
                BLOG
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                className="nav-link"
                href="/contact"
                onClick={[Function]}
              >
                CONTACT
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="hero-inner-container"
  >
    <img
      alt="Drone"
      id="drone-img"
      src="Drone.png"
    />
    <img
      alt="Shadow"
      id="shadow-img"
      src="Shadow.png"
    />
    <h1>
      F-210
    </h1>
    <h3>
      “THE FASTEST DRONE ON THE PLANET”
    </h3>
    <p>
      “The– The Wirecutter
    </p>
  </div>
</section>
`;
